File: solutions/beecrowd/1574/1574.cpp
#include <cstdint>
#include <iostream>
#include <string>

int main() {
    std::int16_t i, t, n, previous_index, position, commands[101];
    std::string command, _junk;

    std::cin >> t;

    while (t--) {
        position = 0;

        std::cin >> n;
        for (i = 1; i <= n; i ++) {
            std::cin >> command;

            if (command == "LEFT") {
                position--;
                commands[i] = -1;
            } else if (command == "RIGHT") {
                position++;
                commands[i] = 1;
            } else {
                std::cin >> _junk >> previous_index;
                position += commands[previous_index];
                commands[i] = commands[previous_index];
            }
        }

        std::cout << position << std::endl;
    }

    return 0;
}

File: solutions/beecrowd/1574/1574.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

commands = [0] * 101

t = int(input())

for _ in range(t):
    position = 0

    n = int(input())
    for i in range(1, n + 1):
        command, *args = input().split()
        match [command, *args]:
            case ['LEFT']:
                position -= 1
                commands[i] = -1
            case ['RIGHT']:
                position += 1
                commands[i] = 1
            case ['SAME', _, index]:
                position += commands[int(index)]
                commands[i] = commands[int(index)]

    output(f'{position}\n')
